packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -118,6 +118,88 @@ enum BottomNavigationBarLandscapeLayout {
 ///
 /// ** See code in examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.1.dart **
 /// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// This example shows [BottomNavigationBar] used in a [Scaffold] Widget with
+/// different interaction patterns. Tapping on the first [BottomNavigationBarItem]
+/// uses the [ScrollController] to animate the [ListView] to the top. The second
+/// and third [BottomNavigationBarItem] change the fontWeight to bold and normal
+/// respectively.
+/// ```dart
```

Comment:
Example code is now hosted in separate files. Can you please put this code in `examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart`? You can follow `examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.1.dart` as a model.

Reply:
Sure, will do it.

Reply:
@guidezpl made the requested changes and added tests.